src/rgw/rgw_op.cc

```diff
@@ -2801,19 +2914,34 @@ void RGWPutObj::execute()
     dout(10) << "v4 auth ok" << dendl;
 
   }
-
   op_ret = store->check_quota(s->bucket_owner.get_id(), s->bucket,
                               user_quota, bucket_quota, s->obj_size);
   if (op_ret < 0) {
     ldout(s->cct, 20) << "second check_quota() returned op_ret=" << op_ret << dendl;
     goto done;
   }
 
-  if (need_calc_md5) {
-    processor->complete_hash(&hash);
-  }
   hash.Final(m);
 
+  if (compression_enabled && compressor->is_compressed()) {
```

Comment:
valgrind complains about this line:
```
==10363== Thread 79:
==10363== Conditional jump or move depends on uninitialised value(s)
==10363==    at 0xEA4CAF: RGWPutObj::execute() (rgw_op.cc:2927)
==10363==    by 0xED84BF: process_request(RGWRados*, RGWREST*, RGWRequest*, RGWStreamIO*, OpsLogSocket*) (rgw_process.cc:181)
==10363==    by 0xD8FD35: civetweb_callback(mg_connection*) (rgw_civetweb_frontend.cc:23)
==10363==    by 0xDADD9A: handle_request (civetweb.c:5799)
==10363==    by 0xDB0482: process_new_connection (civetweb.c:6580)
==10363==    by 0xDB0A2D: worker_thread_run (civetweb.c:6688)
==10363==    by 0xDB0B26: worker_thread (civetweb.c:6723)
==10363==    by 0x1602F619: start_thread (pthread_create.c:334)
==10363==    by 0x1789059C: clone (clone.S:109)
```
..though I'm not exactly sure why. I thought it was just confusing the data dependency between `compression_enabled` and `compressor`, but it gave the same warning when I replaced it with this:
```c++
if (compressor && compressor->is_compressed()) {
```